packages/block-library/src/gallery/edit.js

```diff
@@ -429,24 +429,44 @@ function GalleryEdit( props ) {
 		( img ) => ! img.id && img.url?.indexOf( 'blob:' ) === 0
 	);
 
+	// MediaPlaceholder props are different between web and native hence, we provide a platform-specific set.
+	const mediaPlaceholderProps = Platform.select( {
+		web: {
+			addToGallery: false,
+			disableMediaButtons: imagesUploading,
+			icon: sharedIcon,
+			labels: {
```

Comment:
I think we could share for both web and native the following: `icon`, `labels` and `notices` using the `web` props values. This way we reduce some lines.

```javascript
const mediaPlaceholderProps = Platform.select( {
    web: {
        addToGallery: false,
        disableMediaButtons: imagesUploading,
        value: {},
    },
    native: {
        addToGallery: hasImageIds,
        isAppender: hasImages,
        disableMediaButtons:
            ( hasImages && ! isSelected ) || imagesUploading,
        value: hasImageIds ? images : {},
        autoOpenMediaUpload:
            ! hasImages && isSelected && wasBlockJustInserted,
    },
} );
```

Reply:
Good catch! I've just checked that those props can be shared:

- `icon`, `labels`: These props are only rendered when `isAppender` is `false` or not defined (i.e. when `hasImages` is `false`), so should be safe to use the same code on both platforms.
https://github.com/WordPress/gutenberg/blob/00aea001988f2a8cbd70eeff975d0efebacf1019/packages/block-editor/src/components/media-placeholder/index.native.js#L118-L128

- `notices`: Looks like this prop is not used in the native version of the `MediaPlaceholder` component ([reference](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/media-placeholder/index.native.js)) only on the web, so it should be safe to do the same here.
https://github.com/WordPress/gutenberg/blob/00aea001988f2a8cbd70eeff975d0efebacf1019/packages/block-editor/src/components/media-placeholder/index.native.js#L35-L54
https://github.com/WordPress/gutenberg/blob/00aea001988f2a8cbd70eeff975d0efebacf1019/packages/block-editor/src/components/media-placeholder/index.js#L62

Reply:
I applied the suggestion in https://github.com/WordPress/gutenberg/pull/38238/commits/589d5d8fe8e9192a49bf3a64e192e52294df3e43.